### PR TITLE
speedup optimal snr a bit

### PR DESCRIPTION
**Pull request by ahnitz**
In light of lots of recent issues, I dug up a change I had hacked together to speed up the optimal SNR code a little bit. (~2-3x for SEOBNRv3). This makes it so it can choose the sample rate internally.

### CODE REVIEW

**Comment by spxiwh** [comment]
I think this is going to make bad HAMs a lot worse! Can we discuss on a PyCBC call.

File: bin/pycbc_optimal_snr [hunk 1]
```diff
@@ -107,8 +108,8 @@ if __name__ == '__main__':
                         help='Start frequency of matched-filter integration in Hz (default %(default)s)')
     parser.add_argument('--seg-length', type=float, default=256,
                         help='Segment duration in seconds (default %(default)s)')
-    parser.add_argument('--sample-rate', type=float, default=16384,
-                        help='Data sample rate in Hz (default %(default)s)')
+    parser.add_argument('--sample-rate', type=float,
+                        help='Data sample rate in Hz (default %(default)s). If none, choose dynamically.')
```
Comment (titodalcanton) [inline bin/pycbc_optimal_snr]: The default part needs to be dropped here.

File: bin/pycbc_optimal_snr [hunk 2]
```diff
@@ -170,13 +176,29 @@ if __name__ == '__main__':
     def get_injection(injections, det, injection_time, simulation_id):
         """ Do an injection from the injection XML file, specified by
         IFO and end time"""
-        # leave 4 s of padding at the end for possible ringdown
-        start_time = int(injection_time + 4. - seg_len)
-        strain = TimeSeries(zeros(tlen, dtype=float32), delta_t=delta_t,
+        global tlen, delta_t, sample_rate
+        if optimize_sample_rate:
+            estimate = (pycbc.pnutils.get_final_freq('SEOBNRv2', 1, 100, 1, 1) + 16 ) * 2.1
```
Comment (titodalcanton) [inline bin/pycbc_optimal_snr]: Hmm are the fixed mass and spins intended? I thought the benefit would be in using the injection parameters.

File: bin/pycbc_optimal_snr [hunk 3]
```diff
@@ -208,31 +231,17 @@ if __name__ == '__main__':
...
-    # can be resolved.  This will wastefully run the first injection
-    # twice which could be avoided, but since this is a short-term
-    # kludge anyway I'd prefer to be minimally invasive.
-    # TODO: Remove this when 501 is resolved.
-    # FIXME: https://www.youtube.com/watch?v=DtRNg5uSKQ0
-    for inj in inj_table:
-        ignore, found = compute_optimal_snr(inj, True)
-        if found:
-            break
-
```
Comment (titodalcanton) [inline bin/pycbc_optimal_snr]: image

**Comment by ahnitz** [comment]
Uh... yeah. I don't know how I keep doing this, but I forgot to commit a patch. I think this should be OK now, but I'll do the rerun with Marton's injection file.

**Comment by titodalcanton** [comment]
Hi Alex, could you drop the `%(default)s` in the help of `--sample-rate`? Other than that I think it looks good. Are you ready to merge?

**Comment by spxiwh** [comment]
Any updates on this? If there's nothing new the next time I look over old pull requests then I'll close this ... It's easy to open a new pull request later.

**Comment by duncan-brown** [comment]
@ahnitz poke on this.

**Comment by ahnitz** [comment]
Closing for now. May revisit in the future, but no plans to fix some issues in this, which relate to handling of waveforms with higher modes and precession.

**Pull request closed without merging**